src/pay/payment_producer.py

```diff
@@ -185,12 +182,14 @@ def run(self):
                                 logger.error("This feature is currently not possible using the rpc provider. Please consider changing the provider using the -P flag.")
```

Comment:
This should be enabled using the `expected_reward=True` flag in case of RPC, otherwise follow yaml config